src/Processors/QueryPlan/WindowStep.cpp

```diff
@@ -72,6 +76,11 @@ void WindowStep::transformPipeline(QueryPipelineBuilder & pipeline, const BuildQ
                 input_streams.front().header, output_stream->header, window_description, window_functions);
         });
 
+    if (preserve_num_streams)
```

Comment:
This is a quite dangerous change all the code right now relies on the fact we have only 1 stream after `WindowTrasform`. For example, this can break the order of data between several `WindowTransform`s.

Reply:
@novikd Can you be more specific? I've checked window function over another window function in subquery works fine.

```sql
explain pipeline select sum(number) over (order by number rows between 1 preceding and 1 following)  from (select sum(id) over (order by id rows between 1 preceding and 1 following) as number from window_test
```


```
┌─explain─────────────────────────────────────────────┐
│ (Expression)                                        │
│ ExpressionTransform × 16                            │
│   (Window)                                          │
│   Resize 1 → 16                                     │
│     WindowTransform                                 │
│       (Sorting)                                     │
│       MergingSortedTransform 16 → 1                 │
│         MergeSortingTransform × 16                  │
│           LimitsCheckingTransform × 16              │
│             PartialSortingTransform × 16            │
│               (Expression)                          │
│               ExpressionTransform × 16              │
│                 (Window)                            │
│                 Resize 1 → 16                       │
│                   WindowTransform                   │
│                     (Sorting)                       │
│                     MergingSortedTransform 18 → 1   │
│                       (Expression)                  │
│                       ExpressionTransform × 18      │
│                         (ReadFromMergeTree)         │
│                         MergeTreeInOrder × 18 0 → 1 │
└─────────────────────────────────────────────────────┘
```

Reply:
And multiple windows also works.
```sql
select avg(id) over (order by id rows between 1 preceding and 1 following) as a, avg(value) over (order by id rows between 2 preceding and 1 following) as b from window_test 
```

Reply:
Fixed